### DISCUSSION TITLE
What is the best B&W linux printer?

### DISCUSSION THREAD

**Discussion by mcp292** [OpenPrinting/cups#1011] [Q&A]
What printer has full linux support and rarely ever fails to print what you tell it.

Right now I'm using a Brother printer and jobs sent over consistently get canceled or don't print in full.

Looking for a simple, cheap, reliable, B&W network printer.

**Comment by zdohnal** [comment]
Does your printer have Airprint support? If it does and still does not work, look whether IPP/AirPrint or similar capabilities are enabled on the printer. I have met several models which advertised Airprint, but it was turned off by default. Or, another case, user turned off on purpose and was surprised the printer does not work out of the box...
Then see how your OS handles the printer - if you use a different connection or a different driver (for AirPrint IPP Everywhere or driverless model work, or specific driver for the model, or generic driver for file format which printer supports), you can have bad time.

But to your original question - any printer made after 2010 and claiming Airprint support. Myself I have HP Laserjet from 2010, which although does not advertise Airprint, but it works without any driver and (mostly) flawlessly - I had some issues with fonts on the certain documents, but it got fixed.
Common problem is the current printer (let's say 2015 and newer) is handled as an old printer from < 2010.

**Reply by mcp292** [reply]
Thanks for the reply. 

> Does your printer have Airprint support?...

[There is no mention of an Airprint setting for my printer](https://help.brother-usa.com/app/answers/detail/a_id/152432/~/print-using-airprint).

> Then see how your OS handles the printer...

Nicely done! It looks like it was trying to default to a brother driver. I removed and re-added printer and now it says driverless. Will monitor and report back.

**Reply by mcp292** [reply]
I am having the same problem and noticed a pattern. Printing seems to work fine until I change settings in the print dialog. Namely, print scaling and print quality (e.g., print quality draft -> high).

**Reply by mcp292** [reply]
In other words, problem persists despite the change.

**Reply by zdohnal** [reply]
Can you reproduce via lp command where you provide the print options? You can see option names and values via ```lpoptions -p <printer_name> -l```, where ```<printer_name>``` is seen in the list in ```lpstat -a```.

So IIUC the job gets cancelled or does not print in full if you change print scaling or print quality? What is your OS, CUPS version?

**Reply by mcp292** [reply]
Same exact behavior printing from commandline. Long PDFs get canceled at printer. It seems only one or two jobs work after powering on the printer, then I need to reboot the printer to get through the other documents.

The job usually gets canceled outright, but sometimes an incomplete print will go through where 3 pages or so print and it stops as if complete. At this point I can't say what the cause is. Prints seem to cancel no matter what options I change and whether or not I change any.

When prints fail, the message I see in the Gnome settings view is usually canceled at printer. I have seen some ghostcript error messages. If I come across the ghostscript error again I'll post it. I pulled that string at the time and it didn't lead to a fix.

```
OS: Fedora 40 (Workspace Edition; GNOME)
CUPS: OpenPrinting CUPS 2.4.11
```

**Reply by mcp292** [reply]
I played around a bit in a stuck state (where nothing would print) and found that quitting Document Viewer and re-opening worked. I had to quit; closing the window with others still open and re-opening still got canceled at printer. Document Viewer is usually where I print from, but I have had this happen to me printing through Firefox too, even in this same session.

**Reply by mcp292** [reply]
Did a bit more digging. Tell me if you want to move this thread.

```sh
$ journalctl -f -u cups
Oct 26 21:39:58 fedora cupsd[1290]: REQUEST localhost - - "POST /printers/HL-L2325DW HTTP/1.1" 200 4054720 Print-Job successful-ok
Oct 26 21:40:22 fedora cupsd[1290]: [Job 595] Unable to add document to print job.
Oct 26 21:40:43 fedora cupsd[1290]: [Job 595] Print job canceled at printer.
Oct 26 21:40:43 fedora cupsd[1290]: HL-L2325DW mcp 595 [26/Oct/2024:21:40:43 -0400] total 2 - localhost Printing - Wikipedia - -
Oct 26 21:40:43 fedora cupsd[1290]: [Job 595] Backend dnssd returned status 5 (cancel job)
```

**Reply by mcp292** [reply]
Moved to #1090 with better logs.
